### ISSUE TITLE
Bug: False positive with `@typescript-eslint/no-unnecessary-condition`

### ISSUE THREAD

**Issue by justinsmid** [eslint/eslint#19307]
### Environment

Environment Info:

Node version: v22.8.0
npm version: v10.9.2
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: linux 5.15.0-130-generic

(I run a monorepo setup so I'm guessing the command doesn't quite get the eslint version correctly. We're using eslint version `9.17.0`)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

[ESLint playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=KYDwDg9gTgLgBASwHY2FAZgQwMbDgMQQBtUoAZBAZxgElUBbSmpMAVxgB4AVAPjgG8AUHEQNKALjgBBKFEwBPbj2FxKwTFGwALSdSjIA5oIC%2BgwaEiw46VkmwwEEJNeKkK1OsEYAheQGlgeUolAAoVflEvSgAaVXVNLThjSUISNHdaMWY2Tl5olQBrQIlpWQUOIvkIdDg8wQBKSRk5RV4BFWwnajgiCAB3NABhTDUAZXjtOABeOI1tADoYCDJ%2BoZHgEPqzEShgGFYoZwQxefRXNBCQ46966eUREUrKecoIeg2Qytupvmv6AG1KgBdAD8iwgoxg%2BiQBk24JWAygwzUcOQ2CIrAAJsBKCFeojkcBxnMtPV6vkRFtjEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

<details>
<summary>Configuration</summary>

```
// Simplified config
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error"
  }
}
```
</details>

```ts
export interface FilterListItemsInput<T> {
  items: Array<T>
  search: string
}

export function filterListItemsByKeys<T>(
  { items, search }: FilterListItemsInput<T>,
  keys: Array<keyof T>,
): Array<T> {
  const lowerCaseSearch = search.toLowerCase()

  return items.filter((item) =>
    keys.some((key) => item[key]?.toString().toLowerCase().includes(lowerCaseSearch)),
  )
}
```

### What did you expect to happen?

I expect the code above not to error because `item[key]` **can** be nullish, but the ESLint rule says it can't.

### What actually happened?

`ESLint: Unnecessary optional chain on a non-nullish value.(@typescript-eslint/ no-unnecessary-condition)`

### Link to Minimal Reproducible Example

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=KYDwDg9gTgLgBASwHY2FAZgQwMbDgMQQBtUoAZBAZxgElUBbSmpMAVxgB4AVAPjgG8AUHEQNKALjgBBKFEwBPbj2FxKwTFGwALSdSjIA5oIC%2BgwaEiw46VkmwwEEJNeKkK1OsEYAheQGlgeUolAAoVflEvSgAaVXVNLThjSUISNHdaMWY2Tl5olQBrQIlpWQUOIvkIdDg8wQBKSRk5RV4BFWwnajgiCAB3NABhTDUAZXjtOABeOI1tADoYCDJ%2BoZHgEPqzEShgGFYoZwQxefRXNBCQ46966eUREUrKecoIeg2Qytupvmv6AG1KgBdAD8iwgoxg%2BiQBk24JWAygwzUcOQ2CIrAAJsBKCFeojkcBxnMtPV6vkRFtjEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Tanujkanti4441** [comment]
Hi @justinsmid, thanks for the PR, but rules present in `@typescript-eslint` are maintained by their team, so you have to open this in that repo https://github.com/typescript-eslint/typescript-eslint/issues

**Comment by justinsmid** [comment]
Ah, alright. Will do, thanks!
